table-api/src/main/java/io/deephaven/api/util/NameValidator.java

```diff
@@ -89,7 +89,7 @@ private String validate(String name) {
     }
 
     private static final Set<String> QUERY_LANG_RESERVED_VARIABLE_NAMES =
-            Stream.of("in", "not", "i", "ii", "k").collect(
+            Stream.of("in", "not", "i", "ii", "k", "__rowSet").collect(
```

Comment:
>I checked Formula.java and rowSet appeared to be the only non-mangled item. Though it looks like __lazy __columnsToData and __params are also column names that would fail in formula generation. Those are from the constructor. The column name __temp0 fails which implies __temp[n] is a failing name. Should we add all of these as bad column names, or should we drop __rowSet from the list and just rely on the prefix being an obscure user choice?

I am inclined to take the "`__` is obscure" attitude, but I could see it both ways. I would either add the other items you noted, or take `__rowSet` out of this validation.

*If* we're going to maintain the list, It might be best to add:
1. A static list of internally-used parameters that will conflict with generated formulas (in `DhFormulaColumn` or similar)
2. A unit test of the validator (necessarily in `engine-table`) that verifies that those names aren't allowed.

Maybe there's a better way, involving banning all `__` names, but that feels draconian. Maybe we could use `__DEEPHAVEN_FORMULA_INTERNAL__` as a prefix and ban that?

Reply:
Yeah, I think `__DEEPHAVEN_FORMULA_INTERNAL__` as a prefix is the way to go if we end up needing to ban a prefix.

I think, for now, it makes sense to assume `__` is a sufficiently obscure prefix.